src/main/java/io/github/dgroup/enumerable4j/Enumerable.java

```diff
@@ -302,4 +302,24 @@ default X next(Predicate<X> prd, X alt) {
         return out;
     }
 
+    /**
+     * Returns an enumerable containing first elements of specified size from the enumerable.
+     * @param size The number of elements the enumerable should be limited to.
+     * @return The enumerable.
+     * @throws IllegalArgumentException If the size is negative.
+     */
+    default Enumerable<X> take(long size) {
+        final Enumerable<X> out;
+        if (size < 0) {
+            throw new IllegalArgumentException(Long.toString(size));
+        } else if (size == 0) {
+            out = new Empty<>();
+        } else {
+            out = new Linked<>(
+                this.stream().limit(size).collect(Collectors.toSet())
```

Comment:
why `.toSet()`? 
Use `.toList()` in order to support the same structure

Reply:
idk why i used `.toSet()`, was inconsiderate
thanks for assist